drake/solvers/Optimization.h

```diff
@@ -331,70 +331,99 @@ class DRAKEOPTIMIZATION_EXPORT OptimizationProblem {
     return addQuadraticCost(Q, x_desired, variable_views);
   };
 
-  /** addConstraint
-   * @brief adds a constraint to the program.  method specializations ensure
-   * that the constraint gets added in the right way
+  /** addGenericConstraint
+   *
+   * @brief adds a generic constraint to the program.  This should
+   * only be used if a more specific type of constraint is not
+   * available, as it may require the use of a significantly more
+   * expensive solver.
    */
-  void addConstraint(std::shared_ptr<Constraint> con,
+  void addGenericConstraint(std::shared_ptr<Constraint> con,
                      VariableList const& vars) {
```

Comment:
(As a meta-comment, for this "clang-format" request and other related items ("Function_names", etc.)...)

Fixing pre-existing lint and style errors in order to get a PR merged, combined with multi-day to multi-week PR merge windows, combined with a dozen different developers all changing things at once, is going to turn into merge hell.  (I'd posit that it already has.  Branch-based development is an anti-best-practice.)

Perhaps what we'll need to be doing is that before any functional work touches a pre-existing file (fixing a bug, adding a feature), you have to first stop and fix all the lint and style issues, PR and merge that, and then go on with your functional changes.  However, that isn't what happened for this PR, so I'd argue that because it is strictly better (removing overloading, etc.) it should be merged, and lint fixes done afterward.

Anyway, I have many alternative possible suggestions for a better process, but this tiny textbox margin note on a PR code review is a very unsatisfactory place to discuss them.  Perhaps we can all get together and hash out a consensus proposal next week.